rgbd/src/ros/conversions.cpp

```diff
@@ -112,6 +127,98 @@ bool convert(const cv::Mat& image,
 
     image_cv_bridge.image = img_rect;
     image_cv_bridge.toImageMsg(image_msg);
+
+    return true;
+}
+
+// ----------------------------------------------------------------------------------------------------
+
+bool convert(const rgbd_msgs::RGBDConstPtr& msg, rgbd::Image* image)
```

Comment:
Don't know if I mentioned it in the header, but use smart pointers

Reply:
Because we have a raw pointer in the `ClientRGBD` this is the best interface. Although it is pretty ugly. Changing this API, would add extra copy operations.